libs/src/api/user.js

```diff
@@ -208,86 +208,74 @@ class Users extends Base {
+    await this.creatorNode.updateCreator(userId, metadataFileUUID, txReceipt.blockNumber)
+
+    this.userStateManager.setCurrentUser({ ...oldMetadata, ...newMetadata })
+    return userId
   }
 
   /**
    * Upgrades a user to a creator using their metadata object.
    * This creates a record for that user on the connected creator node.
    * @param {Object} metadata
    */
-  async upgradeToCreator (metadata) {
+  async upgradeToCreator () {
     this.REQUIRES(Services.CREATOR_NODE)
-    this.IS_OBJECT(metadata)
-
-    const oldMetadata = { ...metadata }
-    this._validateUserMetadata(oldMetadata)
 
-    // for now, we only support one user per creator node / libs instance
     const user = this.userStateManager.getCurrentUser()
-    if (!user || user.is_creator) {
-      throw new Error('No current user or existing user is already a creator')
+    if (!user) {
+      throw new Error('No current user')
     }
+    // No-op if the user is already a creator.
+    // Consider them a creator iff they have is_creator=true AND a creator node endpoint
+    if (user.is_creator && user.creator_node_endpoint) return
 
-    oldMetadata.wallet = this.web3Manager.getWalletAddress()
+    const userId = user.user_id
+    const oldMetadata = { ...user }
+    const newMetadata = { ...user }
 
-    const newMetadata = { ...oldMetadata }
+    newMetadata.wallet = this.web3Manager.getWalletAddress()
     newMetadata.is_creator = true
-    if (oldMetadata.creator_node_endpoint) {
-      // Force the new primary creator node to sync from the currently connected node.
-      // TODO: The currently connected node here should be a USER-ONLY node.
-      const newPrimaryCreatorNode = CreatorNodeService.getPrimary(oldMetadata.creator_node_endpoint)
-      if (newPrimaryCreatorNode !== this.creatorNode.getEndpoint()) {
-        await this.creatorNode.forceSync(newPrimaryCreatorNode)
-        await this.creatorNode.setEndpoint(newPrimaryCreatorNode)
-      }
-      // Don't create the creator with an endpoint set, but rather update it once created.
-      // Note: The updateUserOperations only sends transactions with the diff between
-      // metadata and newMetadata, so unsetting creator_node_endpoint here is required.
-      newMetadata.creator_node_endpoint = oldMetadata.creator_node_endpoint
-      oldMetadata.creator_node_endpoint = null
-    } else {
-      // Fallback to the connected creatornode if unset.
-      newMetadata.creator_node_endpoint = this.creatorNode.getEndpoint()
+    newMetadata.creator_node_endpoint = this.creatorNode.getEndpoint()
+
+    // Upload new metadata
+    const { metadataMultihash, metadataFileUUID } = await this.creatorNode.uploadCreatorContent(
+      newMetadata
+    )
+    // Update new metadata on chain
+    let updatedMultihashDecoded = Utils.decodeMultihash(metadataMultihash)
+    const { txReceipt } = await this.contracts.UserFactoryClient.updateMultihash(userId, updatedMultihashDecoded.digest)
```

Comment:
good question. @raymondjacobson thoughts? if tx doesn't confirm then txreceipt would never exist. would it make sense to do some validation here before continuing?

Reply:
@raymondjacobson bump

Reply:
Great call!!!!!!!!!!
We should handle txReceipt.status = FALSE here and abandon out of this method. I thought sendTransaction would fail on this case, but i guess not :)

https://web3js.readthedocs.io/en/v1.2.0/web3-eth.html#eth-gettransactionreceipt-return

Reply:
great catch made a trello task https://trello.com/c/dcsySR1i